feat: adding t-shirt sized typography (SDS-2513) #210

bernhard-adobe opened the pull request:
<!-- Summarize your changes in the Title field -->

Covers:
[SDS-2513](https://jira.corp.adobe.com/browse/SDS-2513) Adding t-shirt sized typography. Issue: #210

[SDS-3259](https://jira.corp.adobe.com/browse/SDS-3259 ) Update Spectrum CSS Typography documentation: Spectrum CSS NextJS & legacy documentation

[SDS-4405](https://jira.corp.adobe.com/browse/SDS-4405) Verify Spectrum CSS Typography Updates

[SDS-4403](https://jira.corp.adobe.com/browse/SDS-4403) Split Typography documentation for Spectrum CSS

[SDS-4460](https://jira.corp.adobe.com/browse/SDS-4460) Update Typekit / Adobe Fonts (Thanks @misterbrownlee )

[SDS-4453](https://jira.corp.adobe.com/browse/SDS-4453) adobe-clean-ux support can be removed #416

## Description

0. Updating Spectrum DNA to get t-shirt sized tokens.
1. Modifying processors.js to allow t-shirt sizes. Some minor restructuring of shared methods.
2. Adding t-shirt sized: sizes, margins and color
3. Enabling the debug statement the processors.js for colors and sizes.
4. Splitting Typography pages 
5. Moving old Typography into Depreciated

For validation the following tokens can be searched:

Paperdoc summary of all changes: https://paper.dropbox.com/doc/Typography-Update--AopzJsKrFFzH_O8gexjcxdEdAg-bluAkl9HGke8WKkJvow2N
(updated t-shirt sized): https://git.corp.adobe.com/gist/nbaldwin/3364cf137d8bd50097c3bedbde9218e5

(original t-shirt sized): https://git.corp.adobe.com/Spectrum/spectrum-dna/pull/310


## How and where has this been tested?
 - **How this was tested:** <!-- Using steps in issue #000 -->
 - **Browser(s) and OS(s) this was tested with:** <!-- Chrome 75.0.3770.142 on Win 10 -->

Ran Backstop.js tests a couple of times: Latest test is  https://spectrum-visual-regression.ci.corp.adobe.com/job/spectrum-css-vr-test-asset/157/artifact/spectrum-css/backstop_data/html_report/index.html

Tested on MacOS 10.14.5 (18F203)
on Chrome Version 80.0.3968.0 (Official Build) canary (64-bit)

1. Run: 
```
yarn install 
gulp dev
```

2. Visit locally:

**New pages:**

[Body](http://localhost:3000/docs/typography-body.html)
[Code](http://localhost:3000/docs/typography-code.html)
[Detail](http://localhost:3000/docs/typograph-detail.html)
[Heading](http://localhost:3000/docs/typography-heading.html)

Old pages:
[Typography page](http://localhost:3000//docs/typography.htm)
[Typography International page](http://localhost:3000//docs/typography-international.html)

## Screenshots
<img width="828" alt="Screen Shot 2019-12-12 at 20 02 25" src="https://user-images.githubusercontent.com/52184321/70768526-67707580-1d1a-11ea-9d07-38e5a57fc740.png">
<img width="239" alt="Screen Shot 2019-12-12 at 20 02 19" src="https://user-images.githubusercontent.com/52184321/70768528-68090c00-1d1a-11ea-8745-61a8ed418b5e.png">




## To-do list
<!-- Put an "x" to indicate you've done each of the following -->
- [x] If my change impacts other components, I have tested to make sure they don't break.
- [x] If my change impacts documentation, I have updated the documentation accordingly.
- [x] I have read the [CONTRIBUTING document](/.github/CONTRIBUTING.md).
<!-- If this pull request isn't ready, add any remaining tasks here -->
- [x] This pull request is ready to merge.

bernhard-adobe commented on tools/component-builder/css/processors.js:
```diff
@@ -10,75 +10,272 @@ OF ANY KIND, either express or implied. See the License for the specific languag
+ * Add CSS nodes that are going to be processed with postcss
+ *
+ * @param {string} mixin name of the mixin
+ * @param {string} css css injected in the mixing
```
typo

bernhard-adobe commented on README.md:
```diff
@@ -160,6 +162,31 @@ Building the project will build and launch the project documentation site in you
 See [site generation](site/README.md) and [documentation generation](tools/bundle-builder/docs/README.md) for more information.
 
+
+### Updating DNA Tokens / CSS variables
```
remove this update documentation. 
Moved into separate ticket & issue: #414 41

NateBaldwinDesign commented on components/typography/index.css:
```diff
@@ -13,6 +13,267 @@ governing permissions and limitations under the License.
+
+
+/* T-Shirt sizes for .spectrum-Article */
+.spectrum-Article {
```
Serif is an option independent of an Article. For the update, I'm not sure it makes any sense to retain this old wrapper concept 🤔.  @GarthDB pinging for input.

bernhard-adobe replied:
@GarthDB  @NateBaldwinDesign we have 2 options here:
1. Keep it wrapped in this `.spectrum-Article` wrapper class and duplicate all the rules outside in the root level. This will increase file size.
2. Remove this wrapper class and move all of the classes in the root level.  This will be most efficient in class name reduction. 

Let me know what you prefer.

GarthDB replied:
I'd default to @NateBaldwinDesign on this. The wrapper is a convenience if everything in the article would use the serif if that's not the case, `article` should be a variant and not a wrapper.

bernhard-adobe replied:
Thanks @GarthDB . I will remove the wrapper class and update the documentation.

NateBaldwinDesign commented on components/typography/index.css:
```diff
@@ -13,6 +13,267 @@ governing permissions and limitations under the License.
+
+/* T-Shirt sizes for .spectrum-Article */
+.spectrum-Article {
+  font-family: var(--spectrum-body-xl-text-font-family);
```
The font family here would need to have been `body-serif-xl-text-font-family` since the old "Article" paradigm = the new Serif option. Again, this wrapper class may not be appropriate, but all `serif` classnames should reference a font-family token from a `serif` DNA option

NateBaldwinDesign commented on components/typography/index.css:
```diff
@@ -13,6 +13,267 @@ governing permissions and limitations under the License.
+  @mixin typographyTShirtSizes .spectrum-BodyXS, body-serif-XS {}
+  @mixin typographyTShirtSizes .spectrum-BodyXXS, body-serif-XS {}
+
+  @mixin typographyTShirtSizes .spectrum-HeadingXXXL--light, heading-light-XXXL {}
```
Not sure why we would include sans-serif classnames under "Article"? I expected to see only the serif options in this wrapper class considering the previously mentioned paradigm of the previous typography

bernhard-adobe replied:
whoops, my bad. should be all serif.

NateBaldwinDesign commented on components/typography/index.css:
```diff
@@ -13,6 +13,267 @@ governing permissions and limitations under the License.
+  @mixin typographyTShirtSizes .spectrum-HeadingXL--heavy, heading-heavy-XL {}
+  @mixin typographyTShirtSizes .spectrum-HeadingL--heavy, heading-heavy-L {}
+
+  /* These don't exist */
```
is there an issue already filed to un-comment these when DNA is released with these options?

bernhard-adobe replied:
logged here as a subtask of the typography updates: 

[SDS-4454 Typography heading sizes: missing DNA tokens ](https://jira.corp.adobe.com/browse/SDS-4454)

NateBaldwinDesign commented:
@bernhard-adobe please make sure that all font weights are included in the Typekit for Spectrum-CSS. It appears that weights heavier than **bold** are not included. This screenshot shows Heading XXL default (top) and Heading XXL Strong (bottom). By default, Heading is bold (font weight 700), but for `<strong>`, the font weight is black (font weight 900). DNA is providing the proper weight value, but it does not appear to be rendering.
![image](https://user-images.githubusercontent.com/13972198/69555592-a0f66200-0f60-11ea-91de-0979687d0e11.png)

Looks like this issue exists for the current typography as well.

NateBaldwinDesign commented:
Here is an additional reference comparing font weights as we have them defined in DNA and what is being rendered in Spectrum-CSS. I've added a hideous thick red border to the items that have incorrect font weights in CSS. Again, I believe this is due to the Typekit not pulling in all necessary styles/weights of Adobe Clean.
![image](https://user-images.githubusercontent.com/13972198/69558361-51666500-0f65-11ea-9d11-ffe4d324d702.png)

NateBaldwinDesign commented:
It also appears as though the internationalized fonts are not included either. When reviewing the Han examples, the letterforms "XXL" did not appear correct. Upon inspection I believe that this is not Adobe Clean Han. It would be good if we have all of the font families specified for CJK languages included in Spectrum-CSS's docs site so that we can accurately review the implementation. 

Below is a screenshot of Spectrum-CSS Han example as-is:
![image](https://user-images.githubusercontent.com/13972198/69559838-ee2a0200-0f67-11ea-8cfc-aee3c0adf670.png)

And here is how it would appear when referencing Adobe Clean Han:
![image](https://user-images.githubusercontent.com/13972198/69559862-f84c0080-0f67-11ea-9f9e-1a042377b10d.png)

NateBaldwinDesign commented:
It appears none of the font stacks are coming directly from DNA. This becomes evident in the fonts listed to use for Arabic and Hebrew, which differ from what's defined in DNA. I would like to discuss this file: https://github.com/adobe/spectrum-css/blob/2f4fab674cc10f692a7ab3bcac799af5ad84a5fd/components/commons/fonts.css#L32 and how we can ensure that CSS pulls these fallback font stacks from DNA here: https://git.corp.adobe.com/Spectrum/spectrum-dna/blob/master/data/aliases/StaticAliases.mjs#L59

NateBaldwinDesign commented on components/typography/metadata/typography.yml:
```diff
@@ -154,11 +294,62 @@ examples:
+        <p class="spectrum-DetailXL">DetailXL <em class="spectrum-DetailXL--light">DetailXL Emphasis</em> <strong class="spectrum-DetailL--strong">DetailXL Strong</strong>.</p>
+        <p class="spectrum-DetailL">DetailL <em class="spectrum-DetailL--light">DetailL Emphasis</em> <strong class="spectrum-DetailL--strong">DetailL Strong</strong>.</p>
+        <p class="spectrum-DetailM">DetailL <em class="spectrum-DetailM--light">DetailM Emphasis</em> <strong class="spectrum-DetailL--strong">DetailM Strong</strong>.</p>
+        <p class="spectrum-DetailS">DetailL <em class="spectrum-DetailS--light">DetailS Emphasis</em> <strong class="spectrum-DetailL--strong">DetailS Strong</strong>.</p>
```
Docs incorrectly label examples as "DetailL" for Medium and Small

bernhard-adobe replied:
> It appears none of the font stacks are coming directly from DNA. This becomes evident in the fonts listed to use for Arabic and Hebrew, which differ from what's defined in DNA. I would like to discuss this file:
> 
> https://github.com/adobe/spectrum-css/blob/2f4fab674cc10f692a7ab3bcac799af5ad84a5fd/components/commons/fonts.css#L32
> 
> and how we can ensure that CSS pulls these fallback font stacks from DNA here: https://git.corp.adobe.com/Spectrum/spectrum-dna/blob/master/data/aliases/StaticAliases.mjs#L59

Just  to clarify: arabic & hebrew use myriad and the rest is going to use adobe clean in it's international versions?
I am not sure why those were hard-coded here, maybe @GarthDB  knows more about this.

bernhard-adobe replied:
is there an option for `  --spectrum-font-family-base: 'adobe-clean-ux', 'adobe-clean', 'Source Sans Pro', var(--spectrum-font-fallbacks-sans);` ?

NateBaldwinDesign commented on components/typography/metadata/typography.yml:
```diff
@@ -154,11 +294,62 @@ examples:
+        <p class="spectrum-DetailXL">DetailXL <em class="spectrum-DetailXL--light">DetailXL Emphasis</em> <strong>DetailXL Strong</strong>.</p>
+        <p class="spectrum-DetailL">DetailL <em class="spectrum-DetailL--light">DetailL Emphasis</em> <strong>DetailL Strong</strong>.</p>
+        <p class="spectrum-DetailM">DetailL <em class="spectrum-DetailM--light">DetailM Emphasis</em> <strong>DetailM Strong</strong>.</p>
+        <p class="spectrum-DetailS">DetailL <em class="spectrum-DetailS--light">DetailS Emphasis</em> <strong>DetailS Strong</strong>.</p>
```
Docs incorrectly label examples as "DetailL" for Medium and Small

NateBaldwinDesign commented on components/typography/metadata/typography.yml:
```diff
@@ -154,11 +294,62 @@ examples:
+        <p class="spectrum-DetailXL">DetailXL <em class="spectrum-DetailXL--light">DetailXL Emphasis</em> <strong>DetailXL Strong</strong>.</p>
+        <p class="spectrum-DetailL">DetailL <em class="spectrum-DetailL--light">DetailL Emphasis</em> <strong>DetailL Strong</strong>.</p>
+        <p class="spectrum-DetailM">DetailL <em class="spectrum-DetailM--light">DetailM Emphasis</em> <strong>DetailM Strong</strong>.</p>
+        <p class="spectrum-DetailS">DetailL <em class="spectrum-DetailS--light">DetailS Emphasis</em> <strong>DetailS Strong</strong>.</p>
```
Docs incorrectly label examples as "DetailL" for Medium and Small

NateBaldwinDesign commented on components/typography/metadata/typography.yml:
```diff
@@ -154,11 +294,62 @@ examples:
+        <p class="spectrum-DetailXL">DetailXL <em class="spectrum-DetailXL--light">DetailXL Emphasis</em> <strong>DetailXL Strong</strong>.</p>
+        <p class="spectrum-DetailL">DetailL <em class="spectrum-DetailL--light">DetailL Emphasis</em> <strong>DetailL Strong</strong>.</p>
+        <p class="spectrum-DetailM">DetailL <em class="spectrum-DetailM--light">DetailM Emphasis</em> <strong>DetailM Strong</strong>.</p>
+        <p class="spectrum-DetailS">DetailL <em class="spectrum-DetailS--light">DetailS Emphasis</em> <strong>DetailS Strong</strong>.</p>
```
Docs incorrectly label examples as "DetailL" for Medium and Small

NateBaldwinDesign commented:
Please add examples of "Detail Light" to this section 
![image](https://user-images.githubusercontent.com/13972198/69561281-5d085a80-0f6a-11ea-8f07-ce6f1ee0a5e1.png)

NateBaldwinDesign commented on components/typography/metadata/typography.yml:
```diff
@@ -180,6 +371,17 @@ examples:
+      <code class="spectrum-CodeL">CodeL Text <strong>Strong</strong></code>
+      <code class="spectrum-CodeM">CodeM Text <strong>Strong</strong></code>
+      <code class="spectrum-CodeS">CodeS Text <strong>Strong</strong></code>
+      <code class="spectrum-CodeXS">CodeXS Text <strong>Strong</strong></code>
```
Please add `<em>Emphasis</em>` examples to code

bernhard-adobe commented:
> Here is an additional reference comparing font weights as we have them defined in DNA and what is being rendered in Spectrum-CSS. I've added a hideous thick red border to the items that have incorrect font weights in CSS. Again, I believe this is due to the Typekit not pulling in all necessary styles/weights of Adobe Clean.
> ![image](https://user-images.githubusercontent.com/13972198/69558361-51666500-0f65-11ea-9d11-ffe4d324d702.png)

We will have to ask @misterbrownlee for either making the update or providing us the credentials.
Ticket here: https://jira.corp.adobe.com/browse/SDS-4460

bernhard-adobe commented:
> It also appears as though the internationalized fonts are not included either. When reviewing the Han examples, the letterforms "XXL" did not appear correct. Upon inspection I believe that this is not Adobe Clean Han. It would be good if we have all of the font families specified for CJK languages included in Spectrum-CSS's docs site so that we can accurately review the implementation.
> 
> Below is a screenshot of Spectrum-CSS Han example as-is:
> ![image](https://user-images.githubusercontent.com/13972198/69559838-ee2a0200-0f67-11ea-8cfc-aee3c0adf670.png)
> 
> And here is how it would appear when referencing Adobe Clean Han:
> ![image](https://user-images.githubusercontent.com/13972198/69559862-f84c0080-0f67-11ea-9f9e-1a042377b10d.png)
This looks like another update to the Typekit instance.

GarthDB commented:
Great and extensive work @bernhard-adobe and @NateBaldwinDesign.
![the-best](https://user-images.githubusercontent.com/125516/70473511-7511cb00-1a8e-11ea-9df7-928a864aae46.gif)

NateBaldwinDesign commented:
@bernhard-adobe just to follow-up on the Slack conversation: for `serif`, we should use the classname `spectrum-Typography--serif` as a replacement for `spectrum-Article`.

bernhard-adobe commented:
blocked currently by the Typekit update: https://jira.corp.adobe.com/browse/SDS-4460

lazd commented:
@bernhard-adobe now that the Typekit updated has been made, is there anything else remaining on this PR? If not, we can go ahead and merge and release it tomorrow!

bernhard-adobe commented:
@lazd  This is ready for review. 
We should discuss if we should hide of of that `depreciated` typography as I feel that it might clutter up a bit the sidebar navigation.

lazd commented:
@bernhard-adobe the sidebar clutter is real, but it shouldn't hold up this PR.

@GarthDB, if this is good to go in your opinion, I say merge it!

bernhard-adobe commented on README.md:
```diff
@@ -214,6 +219,9 @@ Both `backstop:test` and `backstop:docker:test` accept arguments to customize yo
 - `npm run backstop:docker:test themes=lightest,light scales=medium,large radio` - Run test for `radio` and its dependents components with color stop as `lightest` and `light` and scale as `medium` and `large`. It means that a single scenario will run 4 times.
 - `npm run backstop:docker:test themes=lightest,light,dark,darkest scales=medium,large` - Run test for all the components with all the color and scale combinations. It means that a single scenario will be run 8 times.
 
+___
```
I thought it would increase readability in this rather large documentation by adding structure. I will remove it.

bernhard-adobe commented:
@lazd @GarthDB  I made the requested changes:
Examples are:
`class="spectrum-Detail spectrum-Detail--XL"`
`class="spectrum-Heading spectrum-Heading-XXXL--light spectrum-Heading--serif"`

lazd commented:
I think it should be `spectrum-Heading spectrum-Heading--XXXL spectrum-Heading--light spectrum-Heading--serif` -- in SUIT, a single `-` denotes a sub-element, and modifiers can only be applied to the component classname.

bernhard-adobe commented:
@lazd I added those SUIT rules for  "regular, serif, heavy, light" as I believe those only modify the font-weight and style.
Mut I would propose to keep the other discrete rules to allow for an easier transition and be a bit more forgiving in case classnames are being missed or follow from an older standard. Let me know what you think.

lazd commented:
@bernhard-adobe are you saying keep support for `spectrum-Heading-XXXL--light` in addition to `spectrum-Heading--light spectrum-Heading--XXXL`? If so, i must disagree, the former does not conform to the naming convention (it implies there is a subelement called `XXXL` and incorrectly places a modifier on the subelement).

Let's make this one clean and have it perfectly conform to our naming convention, that way we don't have to support anything else in the future.

Also, I'm on PTO so I'm away from my work computer, so can you clarify for me that this PR introduces T-shirt sizes for Typography in addition to the existing (twice deprecated) methods? If so, that's good -- we can do another 2.x release with all of the different Typography APIs together, and we can deprecate the old ones once and for all in the new year with a 3.x release.

lazd commented on components/typography/index.css:
```diff
@@ -211,15 +211,55 @@ governing permissions and limitations under the License.
+    font-weight: var(--spectrum-heading-xl-emphasis-text-font-weight);
+  }
+
+  strong, .spectrum-Heading--strong {
```
This selector targets:
1. child elements of `spectrum-Heading` which are the `strong` tag
2. child elements of `spectrum-Heading` which have the classname `spectrum-Heading--strong`

1 is correct, but 2 is wrong. `spectrum-Heading--strong` is a modifier that should be applied to the outer element, `spectrum-Heading`, such that it affects everything inside of it, right? So it should be:

```
strong, &.spectrum-Heading--strong {
```

Right? That is the intent here, unless I'm missing something? Same should apply to the other typography components.

bernhard-adobe replied:
That is not the intend here. 
Elements nested inside `.spectrum-Heading` used to be only strong or em tags. 
@NateBaldwinDesign  requested to have those also be applied to all all kinds of other elements ergo the `.spectrum-Heading--strong` class. 
By adding the `&`on the .spectrum-Heading--strong class we are applying the selector in combination with the parent selector.
The intend here is to have elements with the classname `.spectrum-heading--strong` inside a `spectrum-Heading` container have the same style like the strong tag.

lazd replied:
In that case, that's the wrong naming convention. You're defining it as a sub-element, so it should use a single dash, i.e. `spectrum-Heading-strong`. In addition, it does not need to be nested inside of `spectrum-Heading`, that adds unnecessary specificity.

bernhard-adobe replied:
I have reverted the naming and removed the double dashed. 

I still suggest that we keep this specificity as it is unclear to me how this change would impact for everyone upgrading to Typography V4. 
Again, the intent of this PR is to implement Typography V4 from a CSS perspective without impacting DOM structure or requiring DOM rewrite.s

bernhard-adobe commented:
I think we should consider supporting the `spectrum-Heading-XXXL--light` for legacy reasons. When converting over to the new Typography updates there will be plenty of cases were our consumers where expecting a single class name only (as it used to be) and might forget adding all the other heading, heading-light and serif classes. It would make using the new typography updates easier. what do you think @GarthDB ?

bernhard-adobe commented:
@lazd  Maybe you should consider reviewing this PR after your PTO? I don't think releasing any of this while you are on PTO is necessary, especially as there is no product need for the Typography updates during this week or shutdown.

bernhard-adobe commented:
@lazd  "can you clarify for me that this PR introduces T-shirt sizes for Typography in addition to the existing (twice deprecated) methods?" Yes. that's why we are keeping the numeric t-shirt sizes and the 2 other legacies. They are not removed from the sidebar. 
 
I believe there is room for debate how far we are going to force the naming conventions and what that would mean for everyone moving to the new Typography V4. 
The intend by me and @NateBaldwinDesign was to only touch the CSS classnames (aka implement the new Typography t-shirt size updates) but not having our user rewrite their HTML structure in order to implement the t-shirt sizes.

lazd commented:
@bernhard-adobe The fact that we're keeping the existing classes for Typography gives us room to properly redefine the API according to the design API and our naming convention. I don't think it makes sense to violate our naming convention to preserve some semblance to a previous violation of our naming convention. If folks are touching their HTML to use the new styles, they'll be in there anyway and can make the necessary changes.

lazd commented:
> @lazd Maybe you should consider reviewing this PR after your PTO? I don't think releasing any of this while you are on PTO is necessary, especially as there is no product need for the Typography updates during this week or shutdown.

Yeah nah we won't release it before shutdown, but I wanted to make sure you had the insight you needed to be unblocked, hence the drive-by PR reviews :)

bernhard-adobe commented:
> Yeah nah we won't release it before shutdown, but I wanted to make sure you had the insight you needed to be unblocked, hence the drive-by PR reviews :)

Thank you, I really appreciate that. It is really not my intention to pull you back into this while on PTO.

So far this PR has been fully unblocked by @GarthDB and @NateBaldwinDesign. Currently I am addressing your feedback which turns out to be a rather large refactor with a lot of opinions and a stronger impact in what it will require to adopt our typography updates. We could chat more about this in January to get a feel if this is the right way to get Typography V4 out of the door. 

I have however made your requested changes.

lazd commented:
@bernhard-adobe thanks for working to address the feedback. You're right, now that we have a design API for Typography that specifically defines the options for each Typography component, it is a rather large refactor to get it in line, but its important to do so for consistency (and important to stay in line with our naming convention in the process). We can discuss it after shutdown if necessary, and I'll hold further review until then.

jianliao commented:
This PR influenced quite a few components. See the difference: https://spectrum-visual-regression.ci.corp.adobe.com/job/spectrum-css-vr-test-asset/140/artifact/spectrum-css/backstop_data/html_report/index.html

bernhard-adobe commented:
we did some more extensive visual testing. the final visual changes are in this Backstop.js test:
https://spectrum-visual-regression.ci.corp.adobe.com/job/spectrum-css-vr-test-asset/157/artifact/spectrum-css/backstop_data/html_report/index.html

bernhard-adobe commented:
latest VR test after rebasing off 2.16.0 

https://spectrum-visual-regression.ci.corp.adobe.com/job/spectrum-css-vr-test-asset/171/artifact/spectrum-css/backstop_data/html_report/index.html

lazd commented:
Things are still getting moved up/down randomly all over the place. I don't think we can merge this change until the cause of the shifting is narrowed down and addressed.

Note the the same shifting was happening in #476 until I backed out the DNA update, so that's where we ought to be looking...

bernhard-adobe commented:
Thanks @lazd 
@jianliao  and I are not sure what in the DNA updates caused this change. Here is one of the pull requests for Typography. https://git.corp.adobe.com/Spectrum/spectrum-dna/pull/310

bernhard-adobe commented:
@lazd I pushed branch `SDS-2513-dna-test` without my typography updates that includes only the update to `"@spectrum/spectrum-dna": "^5.23.0",` from `5.21` and will try to run some test to see what might cause this change.

bernhard-adobe commented:
issues remain, even on DNA  5.24.0.
I am able to reduce the the 1px offset issue at least for CTA buttons by altering this hack from #185 by changing `components/buttons.css https://github.com/adobe/spectrum-css/blob/master/components/button/index.css#L107:
```
  /padding-bottom: calc(var(--spectrum-button-padding-y) + 0.5px);
  padding-top: calc(var(--spectrum-button-padding-y) - 0.5px);
````
to
````
  padding-bottom: calc(var(--spectrum-button-padding-y) -0.5px);
  padding-top: calc(var(--spectrum-button-padding-y) - 1.5px);
````

URL: http://localhost:3000/docs/button-cta.html

Still looking for a root cause of the issue.
 latest test is: https://spectrum-visual-regression.ci.corp.adobe.com/job/spectrum-css-vr-test-asset/186/artifact/spectrum-css/backstop_data/html_report/index.html
This is for example not produceable in Chrome, either on Retina or non-retina:
http://localhost:3000/docs/breadcrumb.html#title
Screenshot: 
![Screen Shot 2020-01-28 at 18 17 48](https://user-images.githubusercontent.com/52184321/73322915-95dadf00-41fa-11ea-95d8-17d26e35b5f9.png)

lazd commented:
Once the analysis in my comment is double-checked, and #511 is merged, we can merge this!

bernhard-adobe commented:
we maybe should think about announcing this in the appropriate channels when we do a release or maybe before if we feel that this is a breaking change.

lazd commented:
@bernhard-adobe since we kept all of the old classes intact, this is not a breaking change. That said, it's a great idea to annouce that Typography has finally caught up.

Additionally, it's a great idea to make this a breaking change in the future and remove the 2 other deprecated Typography implementations :)

lazd commented:
I am working on correcting the component statuses and links on this PR and will merge shortly.

bernhard-adobe commented:
@lazd branched of this branch and created PR #523 to address some more documentation issues and bugs.
